src/storage/src/hummock/sstable/builder.rs

```diff
@@ -224,12 +231,26 @@ impl<W: SstableWriter> SstableBuilder<W> {
     /// | Block 0 | ... | Block N-1 | N (4B) |
     /// ```
     pub async fn finish(mut self) -> HummockResult<SstableBuilderOutput<W::Output>> {
-        let smallest_key = self.block_metas[0].smallest_key.clone();
-        let largest_key = self.last_full_key.clone();
+        let mut smallest_key = if self.block_metas.is_empty() {
+            vec![]
+        } else {
+            self.block_metas[0].smallest_key.clone()
+        };
+        let mut largest_key = self.last_full_key.clone();
 
         self.build_block().await?;
         let meta_offset = self.writer.data_len() as u64;
-        assert!(!smallest_key.is_empty());
+        for (start_key, end_user_key) in &self.range_tombstones {
```

Comment:
My guess is that we need the delete range epoch. Can we use a more structural way to store the epoch instead of embeding it in the start key? For example, have a DeletedRange struct:
```rust
struct DeletedRange {
  start_user_key: Vec<u8>,
  end_user_key: Vec<u8>,
  epoch: u64
}
```

Reply:
Just realized that this is exactly DeleteRangeTombstone. Can we use it instead?

Reply:
`start_key is a full key with epoch and the end_key is a user key?`

I have the same doubt about this

Reply:
Fixed.